pages/index.js

```diff
@@ -167,13 +186,21 @@ export async function getServerSideProps({ req }) {
       await request.getObject("/search/?type=Sample&limit=0")
     ).optional();
 
-    return {
-      props: {
-        fileSets: results["@graph"] || [],
-        fileCount: fileResults?.total || 0,
-        sampleCount: sampleResults?.total || 0,
-      },
+    const props = {
+      fileSets: results["@graph"] || [],
+      fileCount: fileResults?.total || 0,
+      sampleCount: sampleResults?.total || 0,
     };
+
+    try {
+      // Store props in cache (expires in 1 hour).
+      console.log("Setting homepage props in cache");
+      await cache.set("homePageProps", JSON.stringify(props), { EX: 3600 });
```

Comment:
Maybe have a shim function that accepts unstringified props and has a default expiry time. That way this and future clients can just call something like `cache("key", data)` which is sort of how we handle sessionstorage and localstorage.

Reply:
This is good idea but maybe will defer for now.

Would be good if you got used to using client directly since there are a lot of other commands, e.g.: https://www.npmjs.com/package/redis#redis-commands

For example `setEx` is another way to set something with expiration. 

Also it's possible to set strings that are not `JSON.stringified`, which might sometimes be desirable? The two ways give slightly different answers.

```
> await c.set('a', JSON.stringify('b'))
'OK'
> await c.get('a')
'"b"'
> await c.set('a', 'b')
'OK'
> await c.get('a')
'b'
```